src/Events/Custom_Tables/V1/WP_Query/Monitors/Query_Monitor.php

```diff
@@ -25,6 +25,15 @@
 trait Query_Monitor {
 	use With_WP_Query_Introspection;
 
+	/**
+	 * A list of possible modifiers implementations.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	private $implementations = [];
```

Comment:
I would not define this property in the trait. It forces complexity, in the form of classes having to filter to add their own implementations, that is not required in the using classes.

Reply:
Fair point. I would have put it in an abstract class, but the refactor is ballooning. I'll leave it undefined for now, as an optional definition. 

I think typically we should have a base/abstract class these systems should leverage. Helps in scenarios like this, where it's an optional definition.

Reply:
If you define the property in the trait, you cannot override it in the classes using it.
If you think an abstract class would be better, go for it.